tests/standalone_tests/test_healthcheck.py

```diff
@@ -34,3 +48,81 @@ def test_healthcheck_task(
         )
     except Exception as exc:
         pytest.fail(f"Healthcheck failed with error: {exc}")
+
+
+@pytest.fixture(scope="session")
+def controller_config():
+    controller_config = {
+        "rabbitmq": {
+            "user": "user",
+            "password": "password",
+            "vhost": "user_vhost",
+            "celery_tasks_timeout": 10,
+            "celery_run_udf_task_timeout": 30,
+            "celery_tasks_max_retries": 3,
+            "celery_tasks_interval_start": 0,
+            "celery_tasks_interval_step": 0.2,
+            "celery_tasks_interval_max": 0.5,
+        },
+    }
+    return controller_config
+
+
+def get_custom_nodes_addresses_localnode1() -> NodesAddresses:
+    return CustomNodeAddresses([RABBITMQ_LOCALNODE1_ADDR])
+
+
+def get_custom_nodes_addresses_localnodetmp() -> NodesAddresses:
+    return CustomNodeAddresses([RABBITMQ_LOCALNODETMP_ADDR])
+
+
+@pytest.fixture(autouse=True, scope="function")
+def patch_nodes_addresses():
+    with patch.object(
+        NodesAddressesFactory,
+        "get_nodes_addresses",
+    ) as patched:
+        yield patched
+
+
+@pytest.fixture(scope="function")
+def node_landscape_aggregator(controller_config):
+    controller_config = AttrDict(controller_config)
+
+    node_landscape_aggregator = NodeLandscapeAggregator(
+        logger=ctrl_logger.get_background_service_logger(),
+        update_interval=0,
+        tasks_timeout=controller_config.rabbitmq.celery_tasks_timeout,
+        run_udf_task_timeout=controller_config.rabbitmq.celery_run_udf_task_timeout,
+        deployment_type=None,
+        localnodes=None,
+    )
+    node_landscape_aggregator.stop()
+    node_landscape_aggregator.keep_updating = False
+    node_landscape_aggregator._nla_registries = _NLARegistries()
+
+    return node_landscape_aggregator
+
+
+@pytest.mark.slow
+def test_healthcheck_success(
+    patch_nodes_addresses,
+    localnode1_node_service,
+    node_landscape_aggregator,
+):
+    patch_nodes_addresses.side_effect = get_custom_nodes_addresses_localnode1
+
+    try:
+        node_landscape_aggregator.healthcheck()
+    except Exception as exc:
+        pytest.fail(f"Healthcheck failed with exception: {exc}")
+
+
+def test_healthcheck_fail(
+    patch_nodes_addresses,
+    node_landscape_aggregator,
+):
```

Comment:
Clarify the reason for choosing localnodetmp over localnode1 in the failure scenario?

In the test_healthcheck, if we were to remove the localnode1_node_service, would it result in the same behavior observed in the failure case when using localnodetmp?

Reply:
I use localnodetmp since that won't be running for sure.

Reply:
Just in case the localnode1 runs or the background?